### PR TITLE
implementing JENKINS-31125

### PR DESCRIPTION
**Pull request by jarosite**
https://issues.jenkins-ci.org/browse/JENKINS-31125

### CODE REVIEW

**Comment by cyrille-leclerc** [comment]
This looks great. It could make sense to move the code of the drop down list of regions to https://github.com/jenkinsci/aws-java-sdk-plugin as all the AWS plugin would benefit of this selector of AWS Region.

**Comment by jarosite** [comment]
Not sure if it is right idea, because ECS available only in 6 regions and I want to create ECS clouds in few. So general selector of aws region would not play well for me.

**Comment by jarosite** [comment]
Hi All, any chance to review and merge?

**Comment by cyrille-leclerc** [comment]
:+1: looks good but I did not test against an ECS cluster.

@ndeloof can you please look at this?

File: src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSCloud.java [hunk 1]
```diff
@@ -276,6 +295,17 @@ public String getDisplayName() {
             return Messages.DisplayName();
         }
 
+        public ListBoxModel doFillAwsRegionItems() {
```
Comment (ndeloof) [inline src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSCloud.java]: Amazon recommend to query the API for available regions, not use a hard-coded enum. See [ec2-plugin](https://github.com/jenkinsci/ec2-plugin/blob/6b286b185ba41efc33ab8558a6c43969975e6238/src/main/java/hudson/plugins/ec2/AmazonEC2Cloud.java#L158) for sample implementation.

Comment (jarosite) [inline src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSCloud.java]: Yeh, I know, but there no such option for ECS. Even more, ec2 availability do not match to ECS availability.

**Comment by ndeloof** [comment]
already fixed by https://github.com/jenkinsci/amazon-ecs-plugin/pull/7

**Pull request closed without merging**